drivers/storage/ide/atapi/enum.c

```diff
@@ -0,0 +1,656 @@
+/*
+ * PROJECT:     ReactOS ATA Port Driver
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     ATA bus enumeration
+ * COPYRIGHT:   Copyright 2024-2025 Dmitry Borisov <di.sean@protonmail.com>
+ */
+
+/* INCLUDES *******************************************************************/
+
+#include "atapi.h"
+
+/* FUNCTIONS ******************************************************************/
+
+PUCHAR
+AtaCopyIdStringUnsafe(
+    _Out_writes_bytes_all_(Length) PUCHAR Destination,
+    _In_reads_bytes_(Length) PUCHAR Source,
+    _In_ ULONG Length)
+{
+    ULONG i;
+
+    ASSUME((Length >= sizeof(USHORT)) && (Length % sizeof(USHORT)) == 0);
+
+    /* Copy the ATA string and swap it */
+    for (i = 0; i < (Length / sizeof(USHORT)); i += sizeof(USHORT))
+    {
+        Destination[i] = Source[i + 1];
+        Destination[i + 1] = Source[i];
+    }
+
+    return &Destination[i];
```

Comment:
Is this supposed to point at the end (i.e. *after* the buffer)? What's the point of this? It's just gonna be used wrongly (like in `AtaDeviceIsXboxDrive`) and cause a buffer overrun.